apiserver/common/credentialcommon/modelcredential_test.go

```diff
@@ -0,0 +1,348 @@
+
+func (s *ModelCredentialSuite) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+
+	// This is what the test gets from the state.
+	s.machine = createTestMachine("1", "wind-up")
+	s.backend = &mockBackend{
+		Stub: &testing.Stub{},
+		allMachinesFunc: func() ([]credentialcommon.Machine, error) {
+			return []credentialcommon.Machine{s.machine}, nil
+		},
+	}
+
+	// This is what the test gets from the cloud.
+	s.instance = &mockInstance{id: "wind-up"}
+	s.provider = &mockProvider{
+		Stub: &testing.Stub{},
+		allInstancesFunc: func(ctx context.ProviderCallContext) ([]instance.Instance, error) {
+			return []instance.Instance{s.instance}, nil
+		},
+	}
+
+	s.testNewEnvironFunc = func(args environs.OpenParams) (environs.Environ, error) {
+		return &mockEnviron{mockProvider: s.provider}, nil
+	}
+	s.callContext = context.NewCloudCallContext()
+}
+
+func (s *ModelCredentialSuite) TestCheckMachinesSuccess(c *gc.C) {
+	results, err := credentialcommon.CheckMachineInstances(s.backend, s.provider, s.callContext)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{})
+}
+
+func (s *ModelCredentialSuite) TestCheckMachinesInstancesMissing(c *gc.C) {
+	machine1 := createTestMachine("2", "birds")
+	s.backend.allMachinesFunc = func() ([]credentialcommon.Machine, error) {
+		return []credentialcommon.Machine{s.machine, machine1}, nil
+	}
+
+	results, err := credentialcommon.CheckMachineInstances(s.backend, s.provider, s.callContext)
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(results.Results, gc.HasLen, 1)
+	c.Assert(results.Results[0].Error, gc.ErrorMatches, `couldn't find instance "birds" for machine 2`)
+}
+
+func (s *ModelCredentialSuite) TestCheckMachinesExtraInstances(c *gc.C) {
+	instance2 := &mockInstance{id: "analyse"}
+	s.provider.allInstancesFunc = func(ctx context.ProviderCallContext) ([]instance.Instance, error) {
+		return []instance.Instance{s.instance, instance2}, nil
+	}
+
+	results, err := credentialcommon.CheckMachineInstances(s.backend, s.provider, s.callContext)
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(results.Results, gc.HasLen, 1)
+	c.Assert(results.Results[0].Error, gc.ErrorMatches, `no machine with instance "analyse"`)
+}
+
+func (s *ModelCredentialSuite) TestCheckMachinesErrorGettingMachines(c *gc.C) {
+	s.backend.allMachinesFunc = func() ([]credentialcommon.Machine, error) {
+		return nil, errors.New("boom")
+	}
+
+	results, err := credentialcommon.CheckMachineInstances(s.backend, s.provider, s.callContext)
+	c.Assert(err, gc.ErrorMatches, "boom")
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{})
+}
+
+func (s *ModelCredentialSuite) TestCheckMachinesErrorGettingInstances(c *gc.C) {
+	s.provider.allInstancesFunc = func(ctx context.ProviderCallContext) ([]instance.Instance, error) {
+		return nil, errors.New("kaboom")
+	}
+
+	results, err := credentialcommon.CheckMachineInstances(s.backend, s.provider, s.callContext)
+	c.Assert(err, gc.ErrorMatches, "kaboom")
```

Comment:
Ha, ok I did write this code.

I just didn't click that you were talking about this, because I didn't think of this as having anything to do with credentials.